src/connection.rs

```diff
@@ -19,29 +21,36 @@ pub struct Connection {
     pub user:     String,
     pub password: String,
     stream:       RefCell<TcpStream>,
+    query_token:  Cell<u64>,
+}
+
+pub struct QueryResponse {
+    query_token: u64,
+    length:      u32,
+    response:    Json,
 }
 
 /// The response returned by V1_0 of the RethinkDB handshake protocol, after a protocol version has
 /// been successfully set.
 #[derive(Debug,RustcDecodable)]
 struct ProtocolSuccessResponse {
-    success: bool,
+    success:              bool,
```

Comment:
What's going on with the spacing? Is this a monotype alignment thing that's not showing in github correctly?